src/ripple/test/impl/JSONRPCClient.cpp

```diff
@@ -116,6 +118,12 @@ class JSONRPCClient : public AbstractClient
         {
             Json::Value jr;
             jr["method"] = cmd;
+            if (rpc_version_ == 2)
+            {
+                jr["jsonrpc"] = "2.0";
+                jr["ripplerpc"] = "2.0";
+                jr["id"] = 5;
```

Comment:
Would probably want to make the counter atomic just in case we ever run the tests in parallel.  I worry about making the tests so fancy that they become more error prone than what we're testing.  And I fear we may have already approached that boundary.  It is doable.  But I don't think the benefit outweighs the cost.

We are currently checking that the correct id exists in the individual tests (the clients of invoke).  That is also where we check that the other fields exists.